compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MoveCompanionObjectFieldsLowering.kt

```diff
@@ -153,57 +160,12 @@ private class MoveOrCopyCompanionObjectFieldsLowering(val context: CommonBackend
                 isStatic = true
             ).apply {
                 parent = newParent
-                body = oldInitializer.body.transferToNewParent(oldParent, newParent)
+                body = oldInitializer.body
+                    .replaceThisByStaticReference(context, oldParent, oldParent.thisReceiver!!)
+                    .patchDeclarationParents(newParent) as IrBlockBody
             }
         }
 
-    private fun IrBlockBody.transferToNewParent(oldParent: IrClass, newParent: IrClass): IrBlockBody {
```

Comment:
This also fixes a different bug, as the transformed was missing `override fun visitSetVariable`.